From 8260b26f84b500115d80e3237b69d903774611b7 Mon Sep 17 00:00:00 2001
From: Kaan Alsan <alsan@chromium.org>
Date: Thu, 29 Jan 2026 16:42:11 -0800
Subject: [PATCH] Fix crash caused by dragged tabs being targeted by VerticalDraggedTabsContainer

This fixes a race condition when initializing a drag, where a dragged
tab might get targeted. The check to prevent this was looking at the
local cache of VerticalDraggedTabsContainer, but it's possible that it
was not initialized yet.

This is fixed by using the global drag state (through
VerticalTabDragHandler) to check if a tab view is dragging.

Change-Id: If06a3bdd98d9b3b2005855ad8859502a2cf3faf5
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7533527
Reviewed-by: Shibalik Mohapatra <shibalik@chromium.org>
Commit-Queue: Kaan Alsan <alsan@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1576938}
---

--- a/chrome/browser/ui/views/tabs/vertical/vertical_dragged_tabs_container.cc
+++ b/chrome/browser/ui/views/tabs/vertical/vertical_dragged_tabs_container.cc
@@ -243,9 +243,8 @@ views::View* VerticalDraggedTabsContaine
     const views::ProposedLayout& layout,
     const gfx::Point& point) {
   for (const auto& child_layout : layout.child_layouts) {
-    if (child_layout.visible &&
-        !dragging_views_.contains(child_layout.child_view) &&
-        child_layout.bounds.Contains(point)) {
+    if (child_layout.visible && child_layout.bounds.Contains(point) &&
+        !GetDragHandler().IsViewDragging(*child_layout.child_view)) {
       return child_layout.child_view;
     }
   }
